app/controllers/blogs_controller.rb

```diff
@@ -3,13 +3,15 @@
 class BlogsController < ApplicationController
   skip_before_action :authenticate_user!, only: %i[index show]
 
-  before_action :set_blog, only: %i[show edit update destroy]
+  before_action :set_current_user_blog, only: %i[edit update destroy]
 
   def index
     @blogs = Blog.search(params[:term]).published.default_order
   end
 
-  def show; end
+  def show
+    @blog = Blog.where('secret = ? or user_id = ?', false, current_user&.id).find(params[:id])
```

Comment:
この方法でいくなら SQL のキーワードは大文字で書いておくようにしましょう。

```suggestion
    @blog = Blog.where('secret = ? OR user_id = ?', false, current_user&.id).find(params[:id])
```

ところで、Active Record の機能を使うと別の書き方もできそうです。いくつかヒントを書いておくのでトライしてみてください。

- 引数に文字列ではなくハッシュを使う
- 関連の指定に ID ではなくオブジェクトを使う
- OR を Active Record の機能を使って表現する

Reply:
ご指摘いただいたように、Active Record の機能を使って修正しております。

こちら私の理解不足もあったかもしれません。
`Blog.where(...)`のところでクエリが発生すると考えており、`Blog.where(...)`と重ねて書くこと、つまり今回で言うと
`Blog.where(...).or(Blog.where(...))`のようにする修正は行うべきではないと認識してしまっておりました。

`where`はクエリを発行するがDBへのアクセスは行わず、アクセスを行うのは`find`・`find_by`などという理解でよいのでしょうか？

Reply:
> `where`はクエリを発行するがDBへのアクセスは行わず、アクセスを行うのは`find`・`find_by`などという理解でよいのでしょうか？

`find` メソッドと `find_by` メソッドについてはそれでよいと思います。`where` メソッドは、データが実際に必要になるタイミングまでクエリの発行を遅延するメソッドです。クエリを発行しないわけではありません。例えば `rails console` で `Blog.where(secret: true)` のようなコードを実行してみると、`where` メソッドでもクエリの発行が行われていることが分かります。これは、実行結果がコンソールですぐに必要になったからです。このように、クエリの発行を遅延するメソッドとそうでないメソッドがあることを覚えておいてください。

Reply:
ご回答ありがとうございます。
クエリの発行の流れが理解できていなかったので、大変勉強になりました。
今回の場合で言うと、
「`where`→まだクエリを発行しない(遅らせている)→`find`→ここでタイミングがきたからここで発行する」という流れだったのですね。

また、提出物にも記載しているように、コンソールを使って試してみるという考えがすっかり抜けていたので、良い気付きになりました。ありがとうございます。